### PR TITLE
[2.9.x] We are testing upcoming JDK versions here

### PR DESCRIPTION
**Pull request by mkurz**
I want to use this PR to test upcoming JDKs, so I will keep it open "forever" and just push to it from time to time.

<hr>

I set up a custom coursier index file that containts Java 21 (and soon newer) early access builds (because the default coursier index does not contain any):

* https://github.com/mkurz/jvm-index/blob/adoptium_temurin_eaps/index.json

(If you are interested it was done with this [hacked commit](https://github.com/mkurz/jvm-index/commit/6441337a921fcbd7f3fd9d11401400fd19e2080d) so you can generate the json file yourself locally if you want)

You can access the index file yourself and use it if you want:

```sh
$ cs java --available --jvm-index https://github.com/mkurz/jvm-index/raw/adoptium_temurin_eaps/index.json
adoptium:1.21-ea26
adoptium:1.21-ea27
adoptium:1.21-ea28
adoptium:1.21-ea29
adoptium:1.21-ea30
adoptium:1.21-ea31
adoptium:1.21-ea32
adoptium:1.21-ea33
adoptium:1.21-ea34
```

Currently build 35 is the first release candidate for Java 21 LTS, see [the announcement in the mailing list](https://mail.openjdk.org/pipermail/jdk-dev/2023-August/008059.html).

To make this work I needed to add the `jvm-index` argument to the coursier/setup-action workflow:
* https://github.com/coursier/setup-action/pull/465

And of course we need to then integrate that argument into our repo workflows as well:
* https://github.com/playframework/.github/compare/main...jvm-index

### CODE REVIEW

**Comment by mkurz** [comment]
All tests pass with JDK 21 :tada: ! IMHO it's safe to say we can officially support it for the 2.9 release. There will be another JDK21 release candidate (the last one) [end of next week](https://openjdk.org/projects/jdk/21/), I will then test as well, to make it 100% sure.
Ah, of course the only gotcha is the self signed certifacte stuff is not working in Java 21 anymore:

* https://github.com/lightbend/ssl-config/issues/367
* However a possible fix is there already: https://github.com/lightbend/ssl-config/pull/380

Not sure if for development purposes this is really relevant for the majority. Java 17 still works with it so not too bad.

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -156,21 +143,15 @@ jobs:
       - "publish-local"
     uses: playframework/.github/.github/workflows/cmd.yml@v3
     with:
-      java: 17, 11
+      java: 21-ea34
```
Comment (ihostage) [inline .github/workflows/build-test.yml:146]: `21.0.1` released yesterday 😉 
https://github.com/adoptium/temurin21-binaries/releases/tag/jdk-21.0.1%2B12

**Comment by mkurz** [comment]
- Same as in comment https://github.com/playframework/playframework/pull/12139#issuecomment-2136084821

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (mkurz) [reply .github/workflows/build-test.yml:146]: Done
- https://github.com/playframework/playframework/pull/12659
- https://github.com/playframework/playframework/pull/12658
- https://github.com/playframework/playframework/pull/12660

**Pull request closed without merging**